client/src/lifecycleManager.ts

```diff
@@ -56,69 +78,79 @@ class LifecycleManager extends LuigiClientBase {
+        } catch (error) {
+          console.info('unable to parse luigi context data for', key, rawData[key], error);
         }
       }
       this.setCurrentContext(rawData);
     };
 
-    const setAuthData = eventPayload => {
+    const setAuthData = (eventPayload: any) => {
       if (eventPayload) {
         this.authData = eventPayload;
       }
     };
 
-    helpers.addEventListener('luigi.init', e => {
-      setContext(e.data);
-      setAuthData(e.data.authData);
-      helpers.setLuigiCoreDomain(e.origin);
+    helpers.addEventListener('luigi.init', (event: any) => {
+      setContext(event.data);
+      setAuthData(event.data.authData);
+      helpers.setLuigiCoreDomain(event.origin);
       this.luigiInitialized = true;
-      this._notifyInit(e.origin);
+      this._notifyInit(event.origin);
       helpers.sendPostMessageToLuigiCore({ msg: 'luigi.init.ok' });
     });
 
-    helpers.addEventListener('luigi-client.inactive-microfrontend', e => {
-      this._notifyInactive(e.origin);
+    helpers.addEventListener('luigi-client.inactive-microfrontend', (event: any) => {
+      this._notifyInactive(event.origin);
     });
 
-    helpers.addEventListener('luigi.auth.tokenIssued', e => {
-      setAuthData(e.data.authData);
+    helpers.addEventListener('luigi.auth.tokenIssued', (event: any) => {
+      setAuthData(event.data.authData);
     });
 
-    helpers.addEventListener('luigi.navigate', e => {
-      setContext(e.data);
-      if (!this.currentContext.internal.isNavigateBack && !this.currentContext.withoutSync) {
-        const previousHash = window.location.hash;
-        history.replaceState({ luigiInduced: true }, '', e.data.viewUrl);
+    helpers.addEventListener('luigi.navigate', (event: any) => {
+      setContext(event.data);
+
+      if (
+        !(this.currentContext.internal as Record<string, any>)['isNavigateBack'] &&
+        !this.currentContext['withoutSync']
+      ) {
+        const previousHash: string = window.location.hash;
+
+        history.replaceState({ luigiInduced: true }, '', event.data.viewUrl);
         window.dispatchEvent(new PopStateEvent('popstate', { state: 'luiginavigation' }));
+
         if (window.location.hash !== previousHash) {
           window.dispatchEvent(new HashChangeEvent('hashchange'));
         }
       }
+
       // pass additional data to context to enable micro frontend developer to act on internal routing change
-      if (this.currentContext.withoutSync) {
-        Object.assign(this.currentContext.context, {
-          viewUrl: e.data.viewUrl ? e.data.viewUrl : undefined,
-          pathParams: e.data.pathParams ? e.data.pathParams : undefined
+      if (this.currentContext['withoutSync']) {
+        Object.assign(this.currentContext.context as Context, {
+          viewUrl: event.data.viewUrl ? event.data.viewUrl : undefined,
+          pathParams: event.data.pathParams ? event.data.pathParams : undefined
```

Comment:
not sure

Reply:
This is because 'withoutSync' comes from an index signature, as it is not explicitly defined in 'Context' interface